platform/test.rs

```diff
@@ -174,30 +173,77 @@ fn big_data_with_sender_transfer() {
     thread.join().unwrap();
 }
 
-#[test]
-// This test only applies to platforms that need fragmentation.
+macro_rules! create_with_n_fds {
+    ($name:ident, $n:expr, $size:expr) => (
+        #[test]
+        fn $name() {
```

Comment:
Well, I did use a helper function at first; but that means there were a lot of redundant wrappers for each individual test. I didn't like it.

(The macro instantiations admittedly still involve some redundancy for now, due to limitations of the current macro system...)

BTW, I introduced a similar macro already in an earlier PR -- you didn't comment on that one...